From 98b2692dee7bda7a3f5bd77fd460404a1239ae08 Mon Sep 17 00:00:00 2001
From: Hongcheng Xie <hongcheng.xie@intel.com>
Date: Wed, 18 Aug 2021 14:04:29 +0800
Subject: [PATCH] Don't mount cgroup2 in android container

AIC docker will use cgroup mechanism under /sys/fs/cgroup
to config cpu and others resource in container, cgroup2
will mount and overlay the same path, that will lead to
container original config can't take effect.

Change-Id: Ibc45a6b8a5675dbc515b7f8a43d7eecf2084510c
Tracked-On: OAM-98089
Signed-off-by: Hongcheng Xie <hongcheng.xie@intel.com>
---
 libprocessgroup/setup/cgroup_map_write.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libprocessgroup/setup/cgroup_map_write.cpp b/libprocessgroup/setup/cgroup_map_write.cpp
index 25f16a6e9..3921813eb 100644
--- a/libprocessgroup/setup/cgroup_map_write.cpp
+++ b/libprocessgroup/setup/cgroup_map_write.cpp
@@ -236,6 +236,11 @@ static bool SetupCgroup(const CgroupDescriptor& descriptor) {
     if (controller->version() == 2) {
         result = 0;
         if (!strcmp(controller->name(), CGROUPV2_CONTROLLER_NAME)) {
+            bool is_container_mode = GetBoolProperty("ro.container.mode", true);
+            if (is_container_mode) {
+                LOG(INFO) << "Don't mount cgroup2 path: " << controller->path() << " in android container";
+                return false;
+            }
             // /sys/fs/cgroup is created by cgroup2 with specific selinux permissions,
             // try to create again in case the mount point is changed
             if (!Mkdir(controller->path(), 0, "", "")) {
-- 
2.39.0

